packages/page-rpc/src/Rpc/Selection.tsx

```diff
@@ -72,7 +72,7 @@ function Selection ({ queueRpc }: Props): React.ReactElement<Props> {
     (): void => queueRpc({
       rpc,
       values: values
-        .filter(({ value }) => !isNull(value))
+        .filter(({ value }, idx) => !rpc.params[idx].isOptional || !isNull(value))
```

Comment:
Apologies but can you give a brief explanation as to what this is fixing? Thanks again.

Reply:
The `isOptional` setting only really works for last parameters, not for a mix of Option and non-Option parameters.  So the fix keeps null values for `isOptional=false` parameters.

We are using this fix here: https://mainnet-app.polymesh.network/#/rpc
Open the `asset.canTransferGranular` endpoint and just click the submit button (don't need to fill in values).

To see the error try the same endpoint using the Polkadot.js.org app without the fix: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fmainnet-rpc.polymesh.network#/rpc

The error is:
```
1: asset.canTransferGranular: CanTransferGranularReturn
Expected 7 parameters (1 optional), 4 found instead
```

Reply:
Ahh okay, thanks so much for the explanation. Makes sense